typescript/ccip-server/src/services/ProofsService.ts

```diff
@@ -0,0 +1,72 @@
+import { ethers } from 'ethers';
+
+import { TelepathyCcipReadIsmAbi } from '../abis/TelepathyCcipReadIsmAbi';
+
+import { LightClientService } from './LightClientService';
+import { ProofResult, RPCService } from './RPCService';
+import { HandlerDescriptionEnumerated } from './common/HandlerDescriptionEnumerated';
+
+// Service that requests proofs from Succinct and RPC Provider
+class ProofsService extends HandlerDescriptionEnumerated {
+  rpcService: RPCService;
+  lightClientService: LightClientService;
+
+  constructor(
+    readonly lightClientAddress: string,
+    readonly rpcAddress: string,
+    readonly stepFunctionId: string,
+    readonly chainId: string,
+    readonly succinctPlatformUrl: string,
+    readonly succinctPlatformApiKey: string,
+  ) {
```

Comment:
maybe break this out into a couple different objects
```
type SuccinctConfig {
    readonly lightClientAddress: string,
    readonly stepFunctionId: string,
    readonly platformUrl: string,
    readonly apiKey: string,
}
```
and add a dependency on `@hyperlane-xyz/sdk` for `ChainMetadata`